xapian-core/cluster/cluster.cc

```diff
@@ -296,10 +296,10 @@ Point::Point(const TermListGroup &tlg, const Document &document_)
 	string term = *it;
 	double termfreq = tlg.get_termfreq(term);
 
-	/** If term is not a stemmed term or the term indexes only one
-	 *  one document, do not compute TF-IDF scores for them
-	 */
-	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
+	/// If term frequency of the term is lesser than or equal
+	/// to 1 or wdf is lesser than 1, we do not consider this
+	/// term for document vector calculations
+	if (wdf < 1 || termfreq <= 1)
 	    continue;
```

Comment:
The test databases created by the test harness aren't indexed using `TermGenerator` and are indexed by hand but in a way which is approximately like `TermGenerator` with `STEM_ALL` (this is for historical reasons - the harness pre-dates `TermGenerator`, but if we change this it will break many of the existing testcases).

So actually all the terms you are seeing are stemmed, but without a `Z` prefix.

You can create a test database using `TermGenerator` in a more common mode - just use a "generated" database where you pass a function to do the indexing.  The harness will then take care of caching it, only calling that function if it doesn't have a suitable cached database to use.